qbft/spectest/tests/controller/latemsg/late_proposal_past_instance.go

```diff
@@ -37,30 +39,90 @@ func LateProposalPastInstance() tests.SpecTest {
 				),
 			},
 
-			ControllerPostRoot: postRoot,
+			ControllerPostRoot: lateProposalPastInstanceStateComparison(height, nil).Register().Root(),
 		}
 	}
 
+	lateMsg := testingutils.TestingMultiSignerProposalMessageWithHeight([]*bls.SecretKey{ks.Shares[1]}, []types.OperatorID{1}, 2)
+
 	return &tests.ControllerSpecTest{
 		Name: "late proposal past instance",
 		RunInstanceData: []*tests.RunInstanceData{
-			instanceData(qbft.FirstHeight, "24cf697092529cfab3ab06b969d8696692c8bcbb9f41a954f71dc74c3b1d7e97"),
-			instanceData(1, "8aa5464b119518f178d81edf4cea1f4c918f9e084e5262a0e276d3afb00ba620"),
-			instanceData(2, "1799fe0981ae08bde1eae9fef88ef8035f5952974647786287a1a8c36544a5da"),
-			instanceData(3, "2ddf2b8c2f35c8115ddd68120c71e64809bfea6b023ed13e177c2474a95d137d"),
-			instanceData(4, "318816cc8819ad062996704fb4b9990b8088ade0cb4c26816ea0965783bff12e"),
-			instanceData(5, "fd83cdee705de628cf4a9baf9e662c424bf0942c63a680266d7db872d16e9f0a"),
+			instanceData(qbft.FirstHeight),
+			instanceData(1),
+			instanceData(2),
+			instanceData(3),
+			instanceData(4),
+			instanceData(5),
 			{
 				InputValue: []byte{1, 2, 3, 4},
 				InputMessages: []*qbft.SignedMessage{
-					testingutils.TestingMultiSignerRoundChangeMessageWithHeight(
-						[]*bls.SecretKey{ks.Shares[1]},
-						[]types.OperatorID{1},
-						2,
-					),
```

Comment:
Important
this test used wrong late type msg.
Instead of Proposal message was used RoundChange msg

Reply:
Can you open an issue?

Reply:
Good catch!